crates/solver/src/settlement_access_list.rs

```diff
@@ -14,15 +18,95 @@ pub trait AccessListEstimating: Send + Sync {
         tx: &TransactionBuilder<DynTransport>,
     ) -> Result<AccessList> {
         self.estimate_access_lists(std::slice::from_ref(tx))
-            .await
+            .await?
             .into_iter()
             .next()
             .unwrap()
     }
+    /// The function guarantee the same size and order of input and output containers
     async fn estimate_access_lists(
         &self,
         txs: &[TransactionBuilder<DynTransport>],
-    ) -> Vec<Result<AccessList>>;
+    ) -> Result<Vec<Result<AccessList>>>;
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct NodeAccessList {
+    access_list: AccessList,
+}
+
+#[derive(Debug)]
+pub struct NodeApi {
+    url: Url,
+    client: Client,
+    header: HeaderMap, //custom header requires direct usage of client instead of transport
+}
+
+impl NodeApi {
+    pub fn new(url: impl IntoUrl, client: Client, api_key: &str) -> Result<Self> {
```

Comment:
@nlordell the reasoning behind this api_key is to simulate user:pass needed to access the node, but this was only needed for local testing on my machine. After you comment on slack https://gnosisinc.slack.com/archives/C6Z2XNL5Q/p1646642796786589?thread_ts=1645611988.320009&cid=C6Z2XNL5Q it seems like this is not needed and I can use `BatchTransport` here directly ?

Reply:
Fixed.